### PR TITLE
[opentelemetry-collector] update MY_POD_IP with ${env:MY_POD_IP}

### PR DESCRIPTION
**Pull request by JaredTan95**
close https://github.com/open-telemetry/opentelemetry-helm-charts/issues/761

### CODE REVIEW

**Comment by Allex1** [comment]
@JaredTan95 this seems like a breaking change so can you pls bump the minor version of the chart and update the [docs ](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/UPGRADING.md)?

**Comment by JaredTan95** [comment]
> this seems like a breaking change 

Sorry, I don't understand why it's a breaking change. Can you tell me about it? thx~

**Comment by Allex1** [comment]
> Sorry, I don't understand why it's a breaking change. Can you tell me about it? thx~

Reading https://github.com/open-telemetry/opentelemetry-helm-charts/issues/761 seems that `${env:ENV}` syntax is not supported before collector 0.71 . We may have users of this helm chart that overwrite the older version of the collector right ? 
I would make it clear that it will not work in that case. wdyt ?

**Comment by JaredTan95** [comment]
@Allex1 @povilasv thx, got you~

**Review by Allex1 — APPROVED** [review]

**Comment by dmitryax** [comment]
> Reading https://github.com/open-telemetry/opentelemetry-helm-charts/issues/761 seems that ${env:ENV} syntax is not supported before collector 0.71 . We may have users of this helm chart that overwrite the older version of the collector right ?

0.71.0 got only a fix that is unrelated to our use cases. `${env:...}` syntax has been supported for long time. I belive it was added in 2001 :) Not really sure we need the upgrade guidelines

**Review by dmitryax — APPROVED** [review]

File: charts/opentelemetry-collector/UPGRADING.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Upgrade guidelines
 
+## 0.55.0 to 0.55.1
+
+As of v0.55.1 Collector chart use `${env:ENV}` style syntax when getting environment variables and that $`{env:ENV}` syntax is not supported before collector 0.71. If you upgrade collector chart to v0.55.1, you need to make sure your collector version is after than 0.71 (default is v0.76.1).
+
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/UPGRADING.md:6]: Based on @dmitryax comment lets remove this.

**Review by TylerHelmuth — APPROVED** [review]

File: charts/opentelemetry-collector/UPGRADING.md (hunk 1 above)
Comment (JaredTan95) [reply charts/opentelemetry-collector/UPGRADING.md]: done

**Review by dmitryax — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
@mx-psi thanks for the details.  @JaredTan95 sorry, but would you be willing to submit another PR touching up the UPGRADING doc saying that using chart version `0.55.1` or higher requires a collector version `0.71.0` or higher?

**Comment by dmitryax** [comment]
Oh, @mx-psi is right. I am sorry for causing the confusion!

**Pull request merged**